golem-rib/src/type_inference/mod.rs

```diff
@@ -477,7 +477,7 @@ mod type_inference_tests {
         #[test]
         fn test_identifier_type_inference_multiple_re_assign() {
             let rib_expr = r#"
-          let x = 1;
+          let x: u64 = 1;
           let y = x;
```

Comment:
You can see its ambiguous in these expressions as we don't make any function calls, to resolve the types on its own.